kubectl-cloudflow/verify/streamletref.go

```diff
@@ -0,0 +1,97 @@
+package verify
+
+import (
+	"fmt"
+	"github.com/go-akka/configuration"
+	"github.com/lightbend/cloudflow/kubectl-cloudflow/domain"
+	"strings"
+)
+
+// StreamletRef models each element in the streamlets section of a blueprint
+type StreamletRef struct {
+	name string
+	className string
+	problems []BlueprintProblem
+	verified* VerifiedStreamlet
+	metadata* configuration.Config
+	imageId *string
+}
+
+// StreamletDescriptor is just a better domain element typedef
+type StreamletDescriptor domain.Descriptor
+
+func (s* StreamletRef) verify (streamletDescriptors []StreamletDescriptor) StreamletRef {
+	ret := *s
+
+	if !IsDnsLabelCompatible(s.name) {
+		ret.problems = append(ret.problems, InvalidStreamletName{
+				streamletRef: ret.name,
+		})
+	}
+
+	if !CheckFullPatternMatch(ret.className, ClassNamePattern) {
+		ret.problems = append(ret.problems, InvalidStreamletClassName{
+			streamletRef: ret.name,
+			streamletClassName: ret.className,
+		})
+	}
+
+	var found *StreamletDescriptor
+	for _, desc := range streamletDescriptors {
+
+		if desc.ClassName == s.className {
+			found = &desc
+			break
+		}
+	}
+
+	if found != nil {
+		if found.ClassName != "" {
+			ret.className = found.ClassName
+		}
+		ret.verified = &VerifiedStreamlet{s.name, *found}
+	} else {
+		matchingPartially := 0
+		var partialDesc []StreamletDescriptor
+		for _, desc := range streamletDescriptors {
+			if strings.Contains(desc.ClassName, s.className) {
```

Comment:
Need to add a test for `a.b.c.Foo` and `a.b.c.Foo2`, which should not lead to ambiguous error: https://github.com/lightbend/cloudflow/issues/164 Since we now about the issue already rather immediately fix it.

Reply:
Yes I am in the process of adding that, I asked above if we should fix it in this PR.